docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,244 @@
+  - `openshift-baremetal-install create single-node-ignition-config` or
+    `openshift-baremetal-install create ignition-configs` (depending on HA mode)
+
+### Run the matching installer on the assisted-service container
+
+Clusters that have installers that already match the assisted service container's
+architecture could be handled by the assisted-service container as we do today.
+This would require one less image and container per pod, but having the same
+process for every cluster install would be easier to understand and maintain.
+
+### Use RPC over HTTP
+
+[Go's RPC](https://pkg.go.dev/net/rpc@go1.22.3) could be used instead of a direct
+HTTP server (RPC can be hosted over HTTP, but that's not what is being addressed
+here). RPC would make this a simpler change as the code for generating the
+manifests is already contained in a single package, but RPC would be a strange
+choice if we were to move the handling into a truly separate service in the
+future.
+
+### Install multiple libraries on the same image
+
+It may be possible to install both versions of the shared libraries required by
+the installers (libcrypto and libssl?) for FIPS compliance on a single image.
+This would require much less change and should be significantly quicker to
+implement, but it's not clear if these would be possible or supportable.
+This could be achieved by any of the following methods:
+
+1. Create a separate userspace for el8 libraries and chroot when those libraries
+   are required.
+  - This seems a bit complicated and it will likely make our image quite a bit
+    larger than it already is (~1.3G).
+2. Install both versions of the required packages simultaneously.
+  - Not sure if this is possible given that the packages share a name and are
+    only different in version.
+3. Use multi-layer container builds to copy the libraries from an el8 image to a
+   directory on the el9 image and use `LD_PRELOAD` or manipulate `LD_LIBRARY_PATH`
+   to point the el8 installer binaries to the correct libraries.
+
+The approach using chroot worked, but FIPS SMEs said that the container base
+image *must* match the installer for the resulting cluster to be considered
+FIPS-compliant so none of these multi-library options are valid.
+
+### Publish multiple assisted-service images
+
+It's likely that a user installing in FIPS mode will only be interested in
+installing a single OCP version at a time. This means that a given version of
+assisted will need to still support both el8 and el9 FIPS installs, but a single
+deployment of assisted would not.
+
+To handle this the assisted-service image would be built twice; once based on
+el8 and again based on el9. Both images would be released and the operator would
+choose which to deploy based on configuration (likely an annotation since a more
+robust solution would be preferred in the future).
+
+For example, in the case that a user knew they wanted to deploy OCP 4.14 from a
```

Comment:
This assumes that the user who owns the configuration of the management cluster itself is the same user who is provisioning clusters. But often that role is separate. The "Cluster Creator" would need to work with the cluster admin to re-configure assisted-installer to switch it from el8 mode to el9 mode. Then what if they need to switch back? This could become a real hassle.

It would help to have automation that recognizes when the config doesn't work for a desired install operation, but resolving that could be a pain. Even for cases where it is the same person doing both personas, it's inconvenient.

Reply:
This is already something that the application admin is choosing to a degree when they set the available OS images, so I don't think it's too much of a stretch for them to also communicate which OCP versions will be in use.

Reply:
Fair point. So then the time will come when the cluster creator asks the admin to add a new version that requires removal of all other versions. Then they'll have to negotiate an upgrade plan with all stakeholders to get the whole fleet safely over the hump, etc etc. Doesn't that seem like a pain? If we can avoid creating that burden for our customers, we'll all be better-off.